### PR TITLE
fix: enhance audio player error dispatching and teardown

### PR DESCRIPTION
**Pull request by zachraymer**
The audio player will now stop audio when the component is destroyed and the audio player will not dispatch an error when the src is an empty string

## Description

Error handling for audio src will ignore empty strings to avoid error dispatching when component is mounted but src has  not yet been set. The component will also stop any playing audio when teardown occurs, due to no native way to destroy audio sources in browser.
## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by cmullenx — COMMENTED** [review]
Did we decide not to the conditional tooltip that Pronoot requested?

**Comment by zachraymer** [comment]
> Did we decide not to the conditional tooltip that Pronoot requested?

Oops, just added back, was in the wrong changelist.

**Review by cmullenx — APPROVED** [review]
lgtm

File: web-components/src/[sandbox]/examples/date-time-picker.ts [hunk 1]
```diff
@@ -23,7 +23,6 @@ export const dateTimePickerTemplate = html`
   <h3>date-time-picker with value</h3>
   <md-date-time-picker
     value="2021-02-14T12:00:00-08:00"
-    @date-time-change=${() => console.log("fart")}
```
Comment (cmullenx) [inline web-components/src/[sandbox]/examples/date-time-picker.ts:26]: lol thanks for finding that someone did this and removing

Comment (archana93) [reply web-components/src/[sandbox]/examples/date-time-picker.ts:26]: haha, true.Thanks for catching it.

**Review by archana93 — DISMISSED** [review]

**Review by parsoman — APPROVED** [review]

**Review by parsoman — APPROVED** [review]

**Pull request merged**